ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -187,18 +189,29 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            if (scroll) {
-                this.scrollToTop();
-            }
             // enable any validation errors to be visible
             this.validationRequested = true;
             this.sendSectionAnalytics();
             this.currentSection.validate();
+
+            if (scroll) {
+                this.scrollToTop();
+            }
+
             if (this.currentSection.valid) {
                 this.resetValidationState();
-                this.currentSectionIndex = nextIndex;
-                this.visitedSectionIndexes[nextIndex] = true;
-                this.saveLastVisitedSectionIndex(nextIndex);
+
+                this.isPageBusy.pipe(
+                    tap(_ => this.isLoaded$.next(false)),
```

Comment:
what's the reason for setting isLoaded as false here?

Reply:
what if there is no async actions, then template might be blinked for 250ms?

Reply:
actually now I undertstand the reason, w/o it the content will be updated too soon
feel free to ignore this thread